src/sources/kafka.rs

```diff
@@ -167,10 +174,15 @@ async fn kafka_source(
                 };
                 let mut log = LogEvent::default();
 
-                log.insert(
-                    log_schema().message_key(),
-                    Value::from(Bytes::from(payload.to_owned())),
-                );
+                let message = match decode(Bytes::from(payload.to_owned())) {
+                    Ok(message) => message,
+                    Err(error) => {
+                        error!(message = "Error decoding event.", %error);
```

Comment:
We'll want these to follow the `internal_events` pattern so we can emit a rate-limited log message and a counter metric.

Reply:
Adressed in 1a8537e3d1f190b1dec9fe9e6d094bdcf2c53726.